api/v1alpha1/validation.go

```diff
@@ -384,25 +384,68 @@ func (r CertificateSigningRequest) Validate() []error {
 	return allErrs
 }
 
+func (b *Batch) Validate() []error {
+	if b == nil {
+		return []error{errors.New("a batch in a batch sequence must not be null")}
+	}
+	return b.Selector.Validate()
+}
+
+func (b BatchSequence) Validate() []error {
+	var errs []error
+	for _, batch := range lo.FromPtr(b.Sequence) {
+		errs = append(errs, batch.Validate()...)
+	}
+	return errs
+}
+
+func (r *RolloutDeviceSelection) Validate() []error {
+	var errs []error
+	if r == nil {
+		return nil
+	}
+	i, err := r.ValueByDiscriminator()
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		switch v := i.(type) {
+		case BatchSequence:
+			errs = append(errs, v.Validate()...)
+		}
+	}
+	return errs
+}
+
+func (d *DisruptionBudget) Validate() []error {
+	if d == nil {
+		return nil
+	}
+	if d.MinAvailable == nil && d.MaxUnavailable == nil {
```

Comment:
I noticed that `MinAvailable` is allowed to be 0, whereas `MaxUnavailable` is allowed to be at least 1.

Maybe it would be best to keep the minimum also at least to be 1, otherwise it should be unset?

Reply:
There is a difference if they are defined, and if defined what are their allowed values.
For MinAvailable the minimum allowed is 0 - which means that if there is a single device in a group it can still be rolled out using disruption budget.
For MaxUnavailable the minimum is 1 - which means that we do not allow to define a rule that will make roll out impossible.
Anyway, here we just check that at least one of them is defined if disruption budget is defined.

Reply:
But in the end, setting `minAvailable` to 0 is not applying any restriction, IIUC the system would behave the same as if it was `nil`.

Then it would be possible to define a DB that has only `groupBy` + `minAvailable` which I assume would mean the system will calculate the groups, with no other effect.

Anyway, in the UI we will also accept  `minAvailable=0` so we are aligned to the API.

Reply:
I agree with @celdrake - what is the difference between these two disruption budgets?
{}
{minAvailable: 0}

Reply:
Changed the UI to restrict `minAvailable` >= 1 as well.